docs/dev/limit-in-pagination.md

```diff
@@ -0,0 +1,62 @@
+## Background
+
+Scrolled search is performed by request to `http://localhost:9200/<index>/_search?scroll=<timeout>` endpoint and it has a mandatory field `size`, which defines page size:
+```json
+{
+    "size" : 5
+}
+```
+Regular (non-scrolled/non-paged) search is performed by request to `http://localhost:9200/<index</_search` endpoint and it has not mandatory fields. Parameter `size` though defines maximum number of docs to be returned by search.
+
+## Problem statement
+
+`LIMIT` clause is being converted to `size` by SQL plugin during push down operation.

```

Comment:
We need to properly define what we want to accomplish, but discuss what is currently the problem with the code.  I would change this to talk about LIMIT and size conflict, but also include the use cases:
1. When LIMIT > size
2. When size < LIMIT
3. When size == LIMIT (this is kind of naive case, but maybe just mention it anyways)
Then discuss exactly how we expect the system to behave (which does it return a cursor, when does it NOT return a cursor, where does it break (max_window_size), and does this override the default fetch size. 
If there is anything to add in terms of how the JDBC driver will behave, then include it here.